### ISSUE TITLE
`object-curly-newline`: enforce Allman-style brace placement on TypeScript interface and enum declarations

### ISSUE THREAD

**Issue by mho22** [eslint-stylistic/eslint-stylistic#1193]
The `brace-style` rule does not enforce brace placement on TypeScript `interface` and `enum` declarations. Setting the rule to `allman` has no effect on these constructs.

Config:

```ts
'@stylistic/brace-style': ['error', 'allman']
```

Input :

```ts
interface Baz {
    foo: string;
    bar: number;
}

enum Qux {
    A,
    B
}
```

Expected:

```ts
interface Baz
{
    foo: string;
    bar: number;
}

enum Qux
{
    A,
    B
}
```

But no error is reported, the rule does not visit these nodes.

In the deprecated `@typescript-eslint` rule, it handled this correctly by visiting `TSInterfaceBody`, `TSModuleBlock`, and `TSEnumDeclaration`. When the rule was migrated to `@stylistic`, only `TSModuleBlock` was carried over.

Here's the [@typescript-eslint rule]( https://github.com/typescript-eslint/typescript-eslint/blob/6548240e6733381bf306d0b1c0112654c7ef4cd5/packages/eslint-plugin/src/rules/brace-style.ts) 

PR #823 intentionally removed `TSInterfaceBody` and `TSEnumDeclaration` from brace-style, treating them as "object-like" nodes that should be handled by `object-curly-newline` instead. However, `object-curly-newline` only controls newlines inside the braces, not whether the opening brace is placed on a new line. This means there is currently no rule that can enforce Allman-style brace placement on interface and enum declarations.
